pkg/jobs/jobs_test.go

```diff
@@ -269,3 +348,82 @@ func TestJobs_Get(t *testing.T) {
 		})
 	}
 }
+
+func TestJobs_Wait(t *testing.T) {
+	t.Parallel()
+	type TestCase struct {
+		test           string
+		name           string
+		namespace      string
+		status         string
+		delay          time.Duration
+		expectError    bool
+		expectedResult bool
+		timeout        string
+	}
+
+	testCases := []TestCase{
+		{
+			test:           "wait job complete",
+			name:           "pod-running",
+			namespace:      testNamespace,
+			status:         "Complete",
+			delay:          1 * time.Second,
+			expectError:    false,
+			expectedResult: true,
+			timeout:        "5s",
+		},
+		{
+			test:           "timeout waiting job complete",
+			name:           "pod-running",
+			namespace:      testNamespace,
+			status:         "Complete",
+			delay:          10 * time.Second,
+			expectError:    false,
+			expectedResult: false,
+			timeout:        "5s",
+		},
+			{
+			test:           "wait job failed",
+			name:           "pod-running",
+			namespace:      testNamespace,
+			status:         "Failed",
+			delay:          1 * time.Second,
+			expectError:    true,
+			expectedResult: false,
+			timeout:        "5s",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.test, func(t *testing.T) {
+			client := fake.NewSimpleClientset()
+			watcher := watch.NewFake()
+			client.PrependWatchReactor("jobs", k8stest.DefaultWatchReactor(watcher, nil))
+			fixture := New(client, metav1.ListOptions{}, nil)
+			go func(tc TestCase) {
+				time.Sleep(tc.delay)
+				watcher.Modify(testutils.NewJobWithStatus(tc.name, tc.namespace, tc.status))
+			}(tc)
+
+			result, err := fixture.Wait(WaitOptions{
+				Name:      tc.name,
+				Namespace: tc.namespace,
+				Timeout:   tc.timeout,
+			})
+
+			if !tc.expectError && err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+			if tc.expectError && err == nil {
+				t.Errorf("Expected an error but none returned")
```

Comment:
Minor nitpicks for any subsequent update:

* static string can use t.Error("...")
* the error text should start with lower-case, i.e. expected vs. Expected as this would trigger a linting error once enabled